test/tests/functional/pbs_qsub_remove_files.py

```diff
@@ -99,3 +217,20 @@ def test_qalter_remove_files(self):
                                    id=jid)
         except PbsAlterError as e:
             print str(e)
+
+    def test_qalter_direct_write_error(self):
+        """
+        submit a job and after it starts running alter
+        the job with -Roe and check whether expected
+        error message appears
+        """
+        j = Job(TEST_USER)
+        jid = self.server.submit(j)
+        attribs = {ATTR_R: 'oe'}
+        self.server.expect(JOB, {'job_state': 'R'})
+        try:
+            self.server.alterjob(jid, attribs)
+        except PbsAlterError as e:
+            self.assertTrue(
+                'qalter: Cannot modify attribute while job'
```

Comment:
You might want to remove "qalter: " from the message you are trying to search because it may not be there if PTL is running in API mode.